pkg/scheduler/algorithm/predicates/predicates.go

```diff
@@ -240,11 +240,9 @@ func PodFitsResourcesPredicate(pod *v1.Pod, podRequest *schedulernodeinfo.Resour
 // nodeMatchesNodeSelectorTerms checks if a node's labels satisfy a list of node selector terms,
 // terms are ORed, and an empty list of terms will match nothing.
 func nodeMatchesNodeSelectorTerms(node *v1.Node, nodeSelectorTerms []v1.NodeSelectorTerm) bool {
-	nodeFields := map[string]string{}
-	for k, f := range algorithm.NodeFieldSelectorKeys {
-		nodeFields[k] = f(node)
-	}
-	return v1helper.MatchNodeSelectorTerms(nodeSelectorTerms, labels.Set(node.Labels), fields.Set(nodeFields))
+	return v1helper.MatchNodeSelectorTerms(nodeSelectorTerms, labels.Set(node.Labels), fields.Set{
+		api.ObjectNameField: node.Name,
```

Comment:
:thinking: It looks like we haven't actually documented that we only support "metadata.name" when using field selectors.

Reply:
API validation limits it. We can update the API documentation.